### PR TITLE
Add `discourse.dhall-lang.org` configuration

### PR DESCRIPTION
**Pull request by Gabriel439**
This also includes an SMTP server for `mail.dhall-lang.org` necessary to
send activation e-mails.  This currently just has one account for
`discourse@dhall-lang.org`

### CODE REVIEW

**Comment by joneshf** [comment]
I signed up for an account about 10 minutes ago, but haven't received an activation email–neither in the inbox nor in spam. Is it possible the server isn't sending them out?

**Comment by kammitama5** [comment]
I got an activation mail
dhall

**Comment by Gabriel439** [comment]
Let me debug further.  I think I can fix this

**Comment by Gabriel439** [comment]
@joneshf: Try again now

**Comment by joneshf** [comment]
That works. Thanks!

**Comment by Gabriel439** [comment]
I also just added support for GitHub/Google/Twitter-based logins, too

**Comment by f-f** [comment]
Thanks @Gabriel439!

It looks like I'm not able to find the "Login with GitHub" button..this is how it looks like for me:
image

File: nixops/logical.nix [hunk 1]
```diff
@@ -164,6 +199,24 @@
...
+          forceSSL = true;
+
+          extraConfig = ''
+            proxy_set_header Host $host;
+            proxy_set_header X-Real-IP $remote_addr;
+            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+            proxy_set_header X-Forwarded-Proto $scheme;
+            proxy_set_header X-Forwarded-Host $host;
+            proxy_set_header X-Forwarded-Server $host;
+            proxy_set_header Accept-Encoding "";
```
Comment (ocharles) [inline nixops/logical.nix]: FWIW, I think `recommendedProxySettings = true` will set all of this for you.

**Pull request merged**